cmd/commands/product-release.go

```diff
@@ -0,0 +1,245 @@
+		pageLimit         int
+		statusList        string
+		promotionFlowList string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all the pipelines",
+		Args:  cobra.NoArgs,
+		Example: util.Doc(`
+			<BIN> product-release list --product <product>
+		`),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := cmd.Context()
+
+			releaseStatus, err := ToProductReleaseStatus(statusList)
+			promotionFlows, err := ToPromotionFlows(promotionFlowList)
+			if err != nil {
+				return err
+			}
+			filterArgs := platmodel.ProductReleaseFiltersArgs{
+				Statuses:       releaseStatus,
+				PromotionFlows: promotionFlows,
+			}
+			return RunProductReleaseList(ctx, filterArgs, productName, page, pageLimit)
+		},
+	}
+
+	cmd.Flags().StringVarP(&statusList, "status", "s", "", "Filter by statuses")
+	cmd.Flags().StringVar(&promotionFlowList, "promotion-flows", "", "Filter by promotion flows")
+	cmd.Flags().IntVar(&page, "page", 1, "page number")
+	cmd.Flags().IntVar(&pageLimit, "page-limit", 20, "page limit number")
+	cmd.Flags().StringVarP(&productName, "product", "p", "", "product")
+
+	return cmd
+}
+
+func RunProductReleaseList(ctx context.Context, filterArgs platmodel.ProductReleaseFiltersArgs, productName string, page int, pageLimit int) error {
+	pagination := platmodel.SlicePaginationArgs{
+		First: &pageLimit,
+	}
+
+	query := `
+	query getProductReleasesList(
+	  $productName: String!
+	  $filters: ProductReleaseFiltersArgs!
+	  $pagination: SlicePaginationArgs
+	) {
+	  productReleases(productName: $productName, filters: $filters, pagination: $pagination) {
+		pageInfo {
+			hasNextPage
+			hasPrevPage
+			startCursor
+			endCursor
+		}
+		edges {
+		  node {
+			releaseId
+			steps {
+			  environmentName
+			  status
+			}
+			status
+		  }
+		}
+	  }
+	}
+  `
+	// add pagination - default for now is last 20
+	variables := map[string]any{
+		"filters":     filterArgs,
+		"productName": productName,
+		"pagination":  pagination,
+	}
+
+	var productReleasesPage map[string]any
+	productReleasesPage, err := client.GraphqlAPI[map[string]any](ctx, cfConfig.NewClient().InternalClient(), query, variables)
+	if err != nil {
+		return err
+	}
+	productReleaseSlice, err := TransformSlice(productReleasesPage)
+	if len(productReleaseSlice.Edges) == 0 {
+		return fmt.Errorf("no product releases found")
+	}
+	resJSON, err := json.MarshalIndent(productReleaseSlice.Edges, "", "\t")
+
+	fmt.Println(string(resJSON))
+	return nil
+}
+
+func mapToSliceInfo(data map[string]interface{}) (*platmodel.SliceInfo, error) {
+	startCursor, _ := data["startCursor"].(string)
+	endCursor, _ := data["endCursor"].(string)
+	hasNextPage, _ := data["hasNextPage"].(bool)
+	hasPrevPage, _ := data["hasPrevPage"].(bool)
+	return &platmodel.SliceInfo{
+		StartCursor: &startCursor,
+		EndCursor:   &endCursor,
+		HasNextPage: hasNextPage,
+		HasPrevPage: hasPrevPage,
+	}, nil
+}
+
+func TransformSlice(slice map[string]any) (*Slice, error) {
```

Comment:
yeah - i don't think you need that at all

Reply:
if the generic graphql call will return a `Slice`, it will do all this for you

Reply:
what would you do instead?